features/steps/v6uatsteps.rb

```diff
@@ -0,0 +1,53 @@
+require 'date'
+
+# Number
+
+Then(/^the payload field "(.+)" is a number(?: for request (\d+))?$/) do |field_path, request_index|
+    value = read_key_path(find_request(request_index)[:body], field_path)
+    assert_true(value.is_a?(Integer) || value.is_a?(Float))
```

Comment:
assert_true is a blunt instrument, as if the step fails the only diagnostic you get if it fails is:
```
Test::Unit::AssertionFailedError (<true> expected but was)
<false>
```

Using `assert_kind_of Numeric, value` would be better here.  If that fails, you get something like:
```
Test::Unit::AssertionFailedError (<"asd"> was expected to be kind_of?)
<Numeric> but was
<String>.
```

(You can use the Interactive Ruby shell `irb` to try these out locally).

Reply:
TIL, thanks.